File: Midterm/MidtermLAB3/MP02/MP02.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class MP02 {  // <-- Class name matches file name exactly

    public static void main(String[] args) {

        String file = "Sample_Data-Prog-2-csv.csv"; // Make sure dataset.csv is in the same folder
        int count = 0;

        try {
            BufferedReader br = new BufferedReader(new FileReader(file));
            String line;

            while ((line = br.readLine()) != null && count < 10) {
                System.out.println(line);  // Print each row
                count++;
            }

            br.close();

        } catch (IOException e) {
            System.out.println("Error reading file: " + e.getMessage());
        }
    }
}

File: MidtermLab1/MidtermLAB3/MP02/MP02.js
const fs = require('fs');
const readline = require('readline');

const file = 'Sample_Data-Prog-2-csv.csv'; // Make sure this file is in the same folder

let count = 0;

// Create a read stream
const rl = readline.createInterface({
input: fs.createReadStream(file),
crlfDelay: Infinity
});

// Read the file line by line
rl.on('line', (line) => {
if (count < 10) {
    console.log(line);
    count++;
} else {
    rl.close(); // Stop reading after 10 lines
}
});

rl.on('close', () => {
console.log('Finished reading first 10 lines.');
});

rl.on('error', (err) => {
console.error('Error reading file:', err.message);
});